### PR TITLE
Protractor tests login page

### PR DESCRIPTION
**Pull request by ocorella**

### CODE REVIEW

File: tests/e2e/conf.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 exports.config = {
   framework: 'jasmine',
   seleniumAddress: 'http://localhost:4444/wd/hub',
-  specs: ['*.spec.js']
+  specs: ['login.spec.js']
```
Comment (joserv2812) [inline tests/e2e/conf.js]: why this configuration?

File: tests/e2e/login.spec.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+describe('Budget Rent a Car', function() {
+  it('should go to carview page with ocorella credentials', function() {
+    browser.get('http://localhost:8100/#/');
```
Comment (joserv2812) [inline tests/e2e/login.spec.js]: always use browser.get('http://localhost:8100/#/'); add base URL(http://localhost:8100/) in the conf.js

File: tests/e2e/login.spec.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    var password = element(by.model('password'));
+    password.sendKeys('ocorella');
+    expect(element(by.css('.button-primary')).isPresent()).toBe(true);
+    element(by.css(".button-primary")).click();
+    browser.sleep(3000); 
+    expect(browser.getCurrentUrl()).toEqual('http://localhost:8100/#/carview');
+  });
+});
+
+describe('Budget Rent a Car', function() {
```
Comment (joserv2812) [inline tests/e2e/login.spec.js]: don't need this describe

File: tests/e2e/login.spec.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    password.sendKeys('abcdefasdasd');
+    expect(element(by.css('.button-primary')).isPresent()).toBe(true);
+    element(by.css(".button-primary")).click();
+    browser.sleep(3000); 
+    expect(element(by.css('.popup')).isPresent()).toBe(true);
+    expect(browser.getCurrentUrl()).toEqual('http://localhost:8100/#/');
+  });
+});
+
+describe('Budget Rent a Car', function() {
```
Comment (joserv2812) [inline tests/e2e/login.spec.js]: don't need this describe

File: www/app/login/login.controller.js [hunk 5]
```diff
@@ -5,14 +5,13 @@
...
 
-    LoginController.$inject = ['$scope', '$state', '$ionicPopup'];
+    LoginController.$inject = ['$scope', '$state', '$ionicPopup','loginFirebaseService'];
 
-    function LoginController($scope, $state, $ionicPopup) {
-      var ctrlScope = $scope;
-      ctrlScope.user = { };
-      //screen.lockOrientation('portrait');
+    function LoginController($scope, $state, $ionicPopup, loginFirebaseService) {
+      var vm = $scope;
```
Comment (aleks279) [inline www/app/login/login.controller.js]: Doing this:

```
var vm = $scope;
vm.something = stuff;
```

Is not different from doing this:

```
$scope.something = stuff;
```

Still a victim of ionic 1.7.12, issue #2499 ?

File: tests/e2e/login.spec.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  it('should stay in login page and show popup when clicking the register <a> element', function() {
```
Comment (joserv2812) [inline tests/e2e/login.spec.js]: create a beforeEach and declare var username, var password and all elements using in the tests

File: tests/e2e/login.spec.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    browser.sleep(3000); 
+    expect(browser.getCurrentUrl()).toEqual('http://localhost:8100/#/carview');
+  });
+});
+
+describe('Budget Rent a Car', function() {
+  it('should stay in login page when using wrong crendetials and show error popup', function() {
+    browser.get('http://localhost:8100/#/');	
+    var username = element(by.model('username'));
+    username.sendKeys('SIRIASNOTIENECREDENCIALES');
```
Comment (aleks279) [inline tests/e2e/login.spec.js]: :laughing:

File: tests/e2e/carExt.spec.js [hunk 8]
```diff
@@ -1,6 +1,6 @@
 describe('Budget Rent a Car', function() {
   it('should go to carInterior page', function() {
-    browser.get('http://localhost:8100/#/carExterior');
+  	browser.get(browser.baseUrl +'carExterior');
```
Comment (joserv2812) [inline tests/e2e/carExt.spec.js]: bad indentation

File: tests/e2e/carInt.spec.js [hunk 9]
```diff
@@ -1,6 +1,6 @@
 describe('Budget Rent a Car', function() {
   it('should go to trunk page', function() {
-    browser.get('http://localhost:8100/#/carInterior');
+	browser.get(browser.baseUrl +'carInterior');
```
Comment (joserv2812) [inline tests/e2e/carInt.spec.js]: bad indentation

**Pull request merged**